rcl/include/rcl/context.h

```diff
@@ -240,6 +240,31 @@ RCL_WARN_UNUSED
 rcl_context_instance_id_t
 rcl_context_get_instance_id(rcl_context_t * context);
 
+/// Returns the context domain id.
+/**
+ * The given context must be non-`NULL` and be valid.
+ * If context is `NULL`, then `0` will be returned.
+ * If context is uninitialized, then it is undefined behavior.
+ *
+ * <hr>
+ * Attribute          | Adherence
+ * ------------------ | -------------
+ * Allocates Memory   | No
+ * Thread-Safe        | No
```

Comment:
@ivanpauno isn't it?

Reply:
Obviously we need to hash out how we document this, but I would say yes here. Even if you tried to init and access at the same time, it would not cause an error, since it is a primitive that can be accessed atomically. But also I'd just say that the precondition prevents calling init and this function at the same time as defined behavior.

Reply:
> Obviously we need to hash out how we document this, but I would say yes here. Even if you tried to init and access at the same time, it would not cause an error, since it is a primitive that can be accessed atomically.

After thinking a bit, I think you're right. But it's unpredictable if you will get `RCL_RET_INVALID_ARGUMENT` as return code or if the function will succeed.

Reply:
See https://github.com/ros2/rcl/pull/718/commits/313fbca5e072e6cc810ffd974f23d269d21210b1.

Reply:
Maybe `actual_domain_id` should be an atomic? to ensure correct memory ordering.

Reply:
It wouldn't harm to make it atomic, as on most systems it will be a noop.